File: sdks/dotnet/src/Croniq.Runner.Sdk/Protocol/PollResponse.cs
using System.Text.Json.Serialization;

namespace Croniq.Runner.Sdk.Protocol;

internal sealed record PollResponse(
    [property: JsonPropertyName("work")] IReadOnlyList<WorkAssignment> Work,
    [property: JsonPropertyName("cancel")] IReadOnlyList<string> Cancel);

File: sdks/java/core/src/main/java/io/croniq/runner/protocol/PollResponse.java
package io.croniq.runner.protocol;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/** POST /v1/work/poll response body. */
public record PollResponse(
        @JsonProperty("work") List<WorkAssignment> work, @JsonProperty("cancel") List<String> cancel) {

    /** Returns an empty response, used when the server returns {@code null} bodies. */
    public static PollResponse empty() {
        return new PollResponse(List.of(), List.of());
    }
}
